src/Microsoft.OData.Core/Evaluation/ODataConventionalEntityMetadataBuilder.cs

```diff
@@ -406,7 +406,7 @@ private Uri ComputeEditLink()
             Uri uri = this.ResourceMetadataContext.Resource.HasNonComputedId ? this.ResourceMetadataContext.Resource.NonComputedId : this.ComputedId;
 
             Debug.Assert(this.ResourceMetadataContext != null && this.ResourceMetadataContext.TypeContext != null, "this.resourceMetadataContext != null && this.resourceMetadataContext.TypeContext != null");
-            if (this.ResourceMetadataContext.ActualResourceTypeName != this.ResourceMetadataContext.TypeContext.NavigationSourceEntityTypeName)
+            if (this.ResourceMetadataContext.ActualResourceTypeName != this.ResourceMetadataContext.TypeContext.NavigationSourceEntityTypeName && uri.Segments.Last() != this.ResourceMetadataContext.ActualResourceTypeName)

```

Comment:
Does OData spec say "~/odata/customers/NS.VipCustomer/NS.SubVipCustomer not valid?

I think there's no such limitation.

Reply:
It might not be invalid, but it is weird and unnecessary and it gives an implementation burden when implementing the endpoints for it. (With the net core library)

Reply:
It is valid, but not generally useful to put multiple *different* type cast segments -- we shouldn't prevent that.  There is no reason to append the same type segment multiple times, and we shouldn't generate such a URL (which is what this PR fixes).